aten/src/ATen/native/cuda/SpectralOps.cpp

```diff
@@ -293,7 +309,18 @@ static const Tensor& _exec_fft(Tensor& out, const Tensor& self, IntArrayRef out_
   c10::optional<CuFFTConfig> uncached_plan;
   const CuFFTConfig * config = nullptr;
 
-  if (plan_cache.max_size() > 0) {
+  // Workaround for gh-63152, gh-58724
+  // Bluestein plans in CUDA 11.1 (cufft 10.3) cannot be re-used
+  bool use_caching = true;
+  if (10300 <= CUFFT_VERSION && CUFFT_VERSION < 10400) {
+    // Only cache plans for transforms with small prime factors
```

Comment:
What about linking to https://docs.nvidia.com/cuda/cufft/index.html#accuracy-and-performance here? 

Also, for non-expert readability, we probably want to say that Bluestein plans are only used when that condition is met and that other plans are reusable, hence the check

Reply:
I've added the link on `has_large_prime_factor` and added a comment about small prime factors being okay.